File: inf/4/INF.4.02_GrafiAlberi/INF.4.02.03_GrafiJavascript/src/graph.js
"use strict";

class GraphNode {
  constructor(v) {
    this.value = v;
    this.neighbors = [];
  }

  addNeighbor(n) {
    this.neighbors.push(n);
  }

  toString() {
    return `GraphNode val=${this.value} neighbors=${this.neighbors.map(v => v.value)}`;
  }
}

class Graph {
  constructor() {
    this.nodes = [];
  }

  addNode(node) {
    this.nodes.push(node);
  }

  toString() {
    return `Graph\n  ${this.nodes.join('\n  ')}`
  }
}

const graph = new Graph();
const nodes = [1,2,3].map(v => new GraphNode(v));
nodes[0].addNeighbor(nodes[1]);
nodes[1].addNeighbor(nodes[0]);
nodes[0].addNeighbor(nodes[2]);
nodes[2].addNeighbor(nodes[0]);
nodes[2].addNeighbor(nodes[2]);
for (let n of nodes) {
  graph.addNode(n);
}
console.log(graph.toString());

File: inf/4/INF.4.02_GrafiAlberi/INF.4.02.02_Grafi/src/Graph.java
import java.util.ArrayList;

public class Graph<T> {
    private ArrayList<GraphNode<T>> nodes;

    public Graph() {
        this.nodes = new ArrayList<>();
    }

    public void add(GraphNode<T> node) {
        this.nodes.add(node);
    }

    public String toString() {
        String s = "Nodes: \n";
        for (GraphNode<T> n : this.nodes) {
            s += " " + n.toString() + "\n";
        }
        return s;
    };
}
